testing/test_response.go

```diff
@@ -0,0 +1,234 @@
+	return r.AssertStatus(http.StatusUnprocessableEntity)
+}
+
+func (r *TestResponseImpl) AssertTooManyRequests() contractstesting.TestResponse {
+	return r.AssertStatus(http.StatusTooManyRequests)
+}
+
+func (r *TestResponseImpl) AssertInternalServerError() contractstesting.TestResponse {
+	return r.AssertStatus(http.StatusInternalServerError)
+}
+
+func (r *TestResponseImpl) AssertServiceUnavailable() contractstesting.TestResponse {
+	return r.AssertStatus(http.StatusServiceUnavailable)
+}
+
+func (r *TestResponseImpl) AssertHeader(headerName, value string) contractstesting.TestResponse {
+	color.Errorln(r.Response.Header)
+	got := r.Response.Header.Get(headerName)
+	assert.NotEmpty(r.T(), got, fmt.Sprintf("Header [%s] not present on response.", headerName))
+	if got != "" {
+		assert.Equal(r.T(), value, got, fmt.Sprintf("Header [%s] was found, but value [%s] does not match [%s].", headerName, got, value))
+	}
+	return r
+}
+
+func (r *TestResponseImpl) AssertHeaderMissing(headerName string) contractstesting.TestResponse {
+	got := r.Response.Header.Get(headerName)
+	assert.Empty(r.T(), got, fmt.Sprintf("Unexpected header [%s] is present on response.", headerName))
+	return r
+}
+
+func (r *TestResponseImpl) AssertCookie(name, value string) contractstesting.TestResponse {
+	cookie := r.getCookie(name)
+	assert.NotNil(r.T(), cookie, fmt.Sprintf("Cookie [%s] not present on response.", name))
+	assert.Equal(r.T(), value, cookie.Value, fmt.Sprintf("Cookie [%s] was found, but value [%s] does not match [%s]", name, cookie.Value, value))
+	return r
+}
+
+func (r *TestResponseImpl) AssertCookieExpired(name string) contractstesting.TestResponse {
+	cookie := r.getCookie(name)
+	assert.NotNil(r.T(), cookie, fmt.Sprintf("Cookie [%s] not present on response.", name))
+
+	expirationTime := carbon.FromStdTime(cookie.Expires)
+
+	assert.True(r.t, !expirationTime.IsZero() && expirationTime.Lt(carbon.Now()), fmt.Sprintf("Cookie [%s] is not expired; it expires at [%s].", name, expirationTime))
+
+	return r
+}
+
+func (r *TestResponseImpl) AssertCookieNotExpired(name string) contractstesting.TestResponse {
+	cookie := r.getCookie(name)
+	assert.NotNil(r.T(), cookie, fmt.Sprintf("Cookie [%s] not present on response.", name))
+
+	expirationTime := carbon.FromStdTime(cookie.Expires)
+
+	assert.True(r.t, expirationTime.IsZero() || expirationTime.Gt(carbon.Now()), fmt.Sprintf("Cookie [%s] is expired; it expired at [%s].", name, expirationTime))
+	return r
+}
+
+func (r *TestResponseImpl) AssertCookieMissing(name string) contractstesting.TestResponse {
+	assert.Nil(r.t, r.getCookie(name), fmt.Sprintf("Cookie [%s] is present on response.", name))
+
+	return r
+}
+
+func (r *TestResponseImpl) T() *testing.T {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.t
+}
+
+func (r *TestResponseImpl) getStatusCode() int {
+	return r.Response.StatusCode
+}
+
+func (r *TestResponseImpl) getContent() (content string, err error) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
```

Comment:
Ditto, the assertions should be run one by one, is there a chance of conflict?

Reply:
If someone runs it concurrently?